luarules/gadgets/unit_maverick_exp.lua

```diff
@@ -20,39 +20,42 @@ local GetUnitExperience = Spring.GetUnitExperience
 local SetUnitMaxRange = Spring.SetUnitMaxRange
 local GetUnitDefID = Spring.GetUnitDefID
 
-local isMaverick = {}
-for udid, ud in pairs(UnitDefs) do
-	if string.find(ud.name, 'armmav') then
-		isMaverick[udid] = true
-	end
-end
-local maverickOriginalRange = WeaponDefNames["armmav_armmav_weapon"].range
-
 local updateList = {}
 
 function gadget:GameFrame(n)
 	for unitID in pairs(updateList) do
 		local curExp = GetUnitExperience(unitID)
         if curExp then
-            local limExp = ((3*curExp)/(1+3*curExp))*0.6
-            local newRange = maverickOriginalRange * ( 1 + limExp )
+			local unitDef = UnitDefs[GetUnitDefID(unitID)]
+
+			local rangeXPScale = unitDef.customParams.rangexpscale
+            local limExp = ((3*curExp)/(1+3*curExp))*rangeXPScale
```

Comment:
Use full words for variable names, `cur` = `current`, `lim` = `limit`, `exp` = `experience`?
It is okay to use `XP` for `experience` as it is a de facto standard term.

Reply:
Have fixed the code that you have referenced and the variable names.